core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/crt/internal/signer/DefaultAwsCrtV4aHttpSigner.java

```diff
@@ -228,7 +228,7 @@ private static SignedRequest doSign(SignRequest<? extends AwsCredentialsIdentity
 
         HttpRequest crtRequest = toRequest(sanitizedRequest, request.payload().orElse(null));
 
-        V4aContext v4aContext = sign(sanitizedRequest, crtRequest, signingConfig);
+        V4aContext v4aContext = sign(requestBuilder.build(), crtRequest, signingConfig);
```

Comment:
can we do 
```
        SdkHttpRequest.Builder requestBuilder = request.request().toBuilder();

        payloadSigner.beforeSigning(requestBuilder, request.payload().orElse(null), signingConfig.getSignedBodyValue());

        // The requestBuilder changes should be done now.
       SdkHttpRequest requestToSign = requestBuilder.build();

        SdkHttpRequest sanitizedRequest = sanitizeRequest(requestToSign);

        HttpRequest crtRequest = toRequest(sanitizedRequest, request.payload().orElse(null));

        V4aContext v4aContext = sign(requestToSign, crtRequest, signingConfig);
```

Reply:
And I know we have the MultiRegionAccessPointEndpointResolutionTest test, but is worth adding a unit test to this module.